### PR TITLE
implemented Unicode operators

### PR DESCRIPTION
**Pull request by Araq**
refs https://github.com/nim-lang/RFCs/issues/388

### CODE REVIEW

File: compiler/lexer.nim [hunk 1]
```diff
@@ -882,15 +882,61 @@ proc endOperator(L: var Lexer, tok: var Token, pos: int,
...
+      elif L.buf[pos+2] == '\155': result = 3 # ⊛
+      elif L.buf[pos+2] == '\158': result = 3 # ⊞
+      elif L.buf[pos+2] == '\159': result = 3 # ⊟
+      elif L.buf[pos+2] == '\160': result = 3 # ⊠
+      elif L.buf[pos+2] == '\161': result = 3 # ⊡
+    elif L.buf[pos+1] == '\152' and L.buf[pos+2] == '\133': result = 3 # ★
+  of '\194':
+    if L.buf[pos+1] == '\177': result = 2 # ±
+  of '\195':
+    if L.buf[pos+1] == '\151': result = 2 # ×
```
Comment (juancarlospaco) [inline compiler/lexer.nim]: I would leave out the ones of len `2` and simplify this whole proc. 🤔

Comment (Araq) [reply compiler/lexer.nim]: I followed the spec as outlined in the RFC.

Comment (Vindaar) [reply compiler/lexer.nim]: What, those two are two of the most important new symbols :confused:.

File: changelog.md [hunk 2]
```diff
@@ -445,6 +445,13 @@ proc mysort(s: seq; cmp: proc(a, b: T): int) {.effectsOf: cmp.}
   To enable the new effect system, use --experimental:strictEffects.
 
 
+- Nim now supports a small subset of Unicode operators as operator symbols.
+  The supported symbols are: "∙ ∘ × ★ ⊗ ⊘ ⊙ ⊛ ⊠ ⊡ ∩ ∧ ⊓ ± ⊕ ⊖ ⊞ ⊟ ∪ ∨ ⊔".
+  To enable this feature use --experimental:unicodeOperators. Note that due
```
Comment (konsumlamm) [inline changelog.md]:
```suggestion
  To enable this feature, use `--experimental:unicodeOperators`. Note that due
```

**Pull request merged**